### PR TITLE
Add copyto! CUDA kernels

### PR DESCRIPTION
**Pull request by sriharshakandala**
- [ ] Code follows the [style guidelines](https://clima.github.io/ClimateMachine.jl/latest/DevDocs/CodeStyle/) OR N/A.
- [ ] Unit tests are included OR N/A.
- [ ] Code is exercised in an integration test OR N/A.
- [ ] Documentation has been added/updated OR N/A.

### CODE REVIEW

**Comment by simonbyrne** [comment]
bors try

File: src/DataLayouts/cuda.jl [hunk 1]
```diff
@@ -37,42 +45,32 @@ Base.similar(
...
     return nothing
 end
 
 function Base.copyto!(
     dest::IJFH{S, Nij},
     bc::Union{IJFH{S, Nij, A}, Base.Broadcast.Broadcasted{IJFHStyle{Nij, A}}},
 ) where {S, Nij, A <: CUDA.CuArray}
-    _, _, _, _, Nh = size(bc)
-    CUDA.@cuda threads = (Nij, Nij) blocks = (Nh,) knl_copyto!(dest, bc)
+    pdest, pbc = parent(dest), parent(bc)
```
Comment (simonbyrne) [inline src/DataLayouts/cuda.jl:60]: this won't work for `Broadcasted` objects.

Comment (charleskawczynski) [reply src/DataLayouts/cuda.jl:60]: Yeah, this is breaking..

**Comment by charleskawczynski** [comment]
Superseded by #1199

**Pull request closed without merging**